test/ts/unit/collateralized/collateralized_contract.ts

```diff
@@ -0,0 +1,265 @@
+    let collateralContract: MockCollateralizedTermsContractContract;
+    let tokenRegistry: TokenRegistryContract;
+    let mockToken: MockERC20TokenContract;
+    let mockDebtRegistry: MockDebtRegistryContract;
+    let mockTokenRegistry: MockTokenRegistryContract;
+
+    // Scenario runnner
+    const registerTermStartRunner = new RegisterTermStartRunner();
+    const returnCollateralRunner = new ReturnCollateralRunner();
+    const seizeCollateralRunner = new SeizeCollateralRunner();
+
+    const CONTRACT_OWNER = ACCOUNTS[0];
+    const COLLATERALIZER = ACCOUNTS[1];
+    const NON_COLLATERALIZER = ACCOUNTS[2];
+    const BENEFICIARY_1 = ACCOUNTS[3];
+    const BENEFICIARY_2 = ACCOUNTS[4];
+    const MOCK_DEBT_KERNEL_ADDRESS = ACCOUNTS[5];
+    const ATTACKER = ACCOUNTS[6];
+
+    const NULL_PARAMETERS = "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+    const TX_DEFAULTS = { from: CONTRACT_OWNER, gas: 4000000 };
+
+    const COLLATERAL_AMOUNT = Units.ether(5);
+
+    before(async () => {
+        tokenRegistry = await TokenRegistryContract.deployed(web3, TX_DEFAULTS);
+
+        mockDebtRegistry = await MockDebtRegistryContract.deployed(web3, TX_DEFAULTS);
+        mockToken = await MockERC20TokenContract.deployed(web3, TX_DEFAULTS);
+        mockTokenRegistry = await MockTokenRegistryContract.deployed(web3, TX_DEFAULTS);
+
+        /*
+        In our test environment, we want to interact with the contract being
+        unit tested as a statically-typed entity. In order to accomplish this,
+        we take the following steps:
+
+          1 - Instantiate an instance of the contract through the Truffle
+              framework.
+          2 - Instantiate an instance of the contract through the Web3 API using
+              the truffle instance's ABI.
+          3 - Use the Web3 contract instance to instantiate a statically-typed
+              version of the contract as handled by ABI-GEN, which generates
+              a contract wrapper with types pulled from the contract's ABI.
+         */
+
+        // Step 1: Instantiate a truffle instance of the contract.
+        const collateralContractTruffle = await mockCollateralizedContract.new(
+            MOCK_DEBT_KERNEL_ADDRESS,
+            mockDebtRegistry.address,
+            mockTokenRegistry.address,
+            { from: CONTRACT_OWNER },
+        );
+
+        // Step 2: Instantiate a web3 instance of the contract.
+        const collateralContractWeb3Contract = web3.eth
+            .contract(mockCollateralizedContract.abi)
+            .at(collateralContractTruffle.address);
+
+        // Step 3: Instantiate a statically-typed version of the contract.
+        collateralContract = new MockCollateralizedTermsContractContract(
+            collateralContractWeb3Contract,
+            TX_DEFAULTS,
+        );
+
+        // Initialize scenario runners
+        registerTermStartRunner.initialize(
+            {
```

Comment:
can we assign these objects to variables and then use them in order to not duplicate them

Reply:
Yup, good catch

Reply:
Done